library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -136,6 +123,20 @@ internal class TransactionPayloadFragment :
         super.onCreateOptionsMenu(menu, inflater)
     }
 
+    private fun shouldShowSaveIcon(transaction: HttpTransaction?) = when {
+        // SAF is not available on pre-Kit Kat so let's hide the icon.
+        (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) -> false
+        (type == TYPE_REQUEST && 0L == (transaction?.requestContentLength ?: 0L)) -> false
+        (type == TYPE_RESPONSE && 0L == (transaction?.responseContentLength ?: 0L)) -> false
+        else -> true
+    }
+
+    private fun shouldShowSearchIcon(transaction: HttpTransaction?) = when {
+        (type == TYPE_REQUEST && true == transaction?.isRequestBodyPlainText) -> true
```

Comment:
It seems that there is a bug - with such condition we will show search icon even for empty request/response. Suggest to add check for `contentLength` here as well.

Reply:
```suggestion
        ((type == TYPE_REQUEST) && (true == transaction?.isRequestBodyPlainText) && (transaction?.requestContentLength ?: 0L)) -> true
```